### ISSUE TITLE
tf.cond() calls execute next_batch() in all branches 

### ISSUE THREAD

**Issue by kdavis-mozilla** [mozilla/DeepSpeech#215]
Merge 08064e7 introduced two `tf.cond()` calls to simplify the code. However, they introduce a significant overhead as they call  `next_batch()` in all branches. 

More specifically, according to the `tf.cond()` documentation, for the simple program
```python
z = tf.mul(a, b)
result = tf.cond(x < y, lambda: tf.add(x, z), lambda: tf.square(y))
```
one has

> If x < y, the tf.add operation will be executed and tf.square operation will not be executed. Since z is needed for at least one branch of the cond, the tf.mul operation is always executed, unconditionally. Although this behavior is consistent with the dataflow model of TensorFlow, it has occasionally surprised some users who expected a lazier semantics.

This implies that for our code
```python
# Fetch the next batch of data
batch_x, batch_x_seq_len, batch_y, batch_y_seq_len = \
tf.cond(tf.less(queue_selector_placeholder, 1),
        lambda: batch_sets[0].next_batch(),
        lambda: tf.cond(tf.less(queue_selector_placeholder, 2),
                        lambda: batch_sets[1].next_batch(),
                        lambda: batch_sets[2].next_batch()))
```
all of the calls

- `batch_sets[0].next_batch()`
- `batch_sets[1].next_batch()`
- `batch_sets[2].next_batch()`

are always executed making our code much slower.

**Comment by kdavis-mozilla** [comment]
Commit cfc36ca reverted the `tf.cond()` code.

**Comment by kdavis-mozilla** [comment]
I guess for the near-term we can consider this solved through commit cfc36ca
